tests/html/extractors/factories/embeddedAttributeJs.test.ts

```diff
@@ -0,0 +1,47 @@
+import { embeddedAttributeJsExtractor } from '../../../../src/html/extractors/factories/embeddedAttributeJs';
+import { HtmlParser } from '../../../../src/html/parser';
+import { JsParser } from '../../../../src/js/parser';
+
+describe('HTML: Attribute Value as Embedded JS Extractor', () => {
+
+    describe('calling js parser', () => {
+
+        let htmlParser: HtmlParser,
+            jsParserMock: JsParser;
+
+        beforeEach(() => {
+            jsParserMock = <any>{
+                parseString: jest.fn()
+            };
+
+            htmlParser = new HtmlParser(undefined!, [
+                embeddedAttributeJsExtractor(jsParserMock, ':title')
+            ]);
+        });
+
+        test('single line', () => {
+            htmlParser.parseString(`<span :title="__('msg id')" title="__('another')">content</span>`, 'foo.html');
+            expect(jsParserMock.parseString).toHaveBeenCalledWith(`__('msg id')`, 'foo.html', {
+                lineNumberStart: 1
+            });
+        });
+
+        test('with lineNumberStart option', () => {
+            htmlParser.parseString(`<span :title="__('msg id')" :class="abc">content</span>`, 'foo.html', { lineNumberStart: 10 });
+            expect(jsParserMock.parseString).toHaveBeenCalledWith(`__('msg id')`, 'foo.html', {
+                lineNumberStart: 10
+            });
+        });
+
```

Comment:
The filter callback needs a test as well

Reply:
Maybe you misunderstood my comment. I would like a test for using a filter function instead of the regex.

Reply:
Hello, I have update the test case
```js
            htmlAttrParser = new HtmlParser(undefined!, [
                embeddedAttributeJsExtractor(e => { return e.name.startsWith(':') }, jsParserMock,)
            ]);

        test('filter attr', () => {
            htmlAttrParser.parseString(`<span :title="__('title')" class="title">Hello</span>`, 'foo.html');
            expect(jsParserMock.parseString).toHaveBeenCalledWith(`__('title')`, 'foo.html', { lineNumberStart: 1 });
        })
```
Is this ok?

Reply:
The test itself is good, but the naming is unclear:
- `htmlAttrParser` vs `htmlParser` doesn't highlight what's different about them
- `'filter attr'` doesn't really say what it's testing either

I would suggest you move the instantiation of `HtmlParser` from the `beforeEach` into the individual `test` blocks and remove the top level `htmlParser` and `htmlAttrParser` variables. That will lead to a slight bit of code duplication, but should be much easier to understand. And don't forget to change the test name as well.

Reply:
Thanks for your patience I would fix it now

Reply:
Great! content wise it looks good now. Just one final thing: please revert back to using single quotes.

Reply:
Updated!